bionic/flow.py

```diff
@@ -541,62 +541,70 @@ def merge(self, flow, keep='error', allow_name_match=False):
 
             if conflict.new_provider is None:
-                conflict.resolve('old', 'no conflicting definition')
+                conflict.resolve('self', 'no conflicting definition')
                 continue
 
             if conflict.name == 'core__flow_name':
-                conflict.resolve('old', 'flow name is never merged')
+                conflict.resolve('self', 'flow name is never merged')
                 continue
 
             if conflict.new_is_default:
-                conflict.resolve('old', 'conflicting definition is default')
+                conflict.resolve('self', 'conflicting definition is default')
                 continue
 
             if conflict.old_is_default:
-                conflict.resolve('new', 'conflicting definition is default')
+                conflict.resolve('arg', 'conflicting definition is default')
                 continue
 
             if conflict.old_protocol is conflict.new_protocol:
                 if conflict.new_is_only_declaration:
                     conflict.resolve(
-                        'old',
+                        'self',
                         'conflicting definition has matching protocol and '
                         'no value')
                     continue
                 elif conflict.old_is_only_declaration:
                     conflict.resolve(
-                        'new',
+                        'arg',
                         'conflicting definition has matching protocol and '
                         'no value')
                     continue
 
             if keep == 'error':
                 raise AlreadyDefinedEntityError(
-                    "Merge failure: Entity %r exists in both old and new "
+                    "Merge failure: Entity %r exists in both self and arg "
                     "flows; use the ``keep`` argument to specify which to "
                     "keep" % conflict.name)
 
+            elif keep == 'self':
+                conflict.resolve('self', 'keep=self')
+                continue
+
+            elif keep == 'arg':
+                conflict.resolve('arg', 'keep=arg')
+                continue
+
             elif keep == 'old':
-                conflict.resolve('old', 'keep=old')
+                conflict.resolve('self', 'keep=old')
                 continue
 
             elif keep == 'new':
-                conflict.resolve('new', 'keep=new')
+                conflict.resolve('arg', 'keep=new')
                 continue
 
             raise ValueError(
-                "Value of ``keep`` must be one of {'error', 'old', 'new'}; "
+                "Value of ``keep`` must be one of {'error', 'self', 'arg'}; "
                 "got %r" % keep)
 
         # For both states, check that each jointly-defined name group is kept
         # or discarded as a whole.
         for state_name, state in [
-                    ('old', old_state),
-                    ('new', new_state),
+                    ('self', old_state),
```

Comment:
WDYT about keeping `old` and `new` for the internal strings, to match `old_state` and `new_state`?  I can see why you don't want `self_is_default`, etc.

Reply:
yeah good point